src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PollingFileChangeToken.cs

```diff
@@ -65,7 +65,7 @@ private DateTime GetLastWriteTimeUtc()
                 catch (IOException) { } // https://github.com/dotnet/runtime/issues/57221
             }
 
-            return lastWriteTimeUtc.Value;
+            return lastWriteTimeUtc;
```

Comment:
Instead of changing multiple lines, can we do this?
```suggestion
            return lastWriteTimeUtc.GetValueOrDefault();
```

Reply:
Then it could affect [this logic](https://github.com/dotnet/runtime/pull/57415/files#diff-f0404a6e281b49fe956eb63b7133b6d334f155f04173ec3c6b333b4fcb9948b7R116) and `hasChanged` will be true, when in reality this might not be the case.

Reply:
you mean for files older that the default value of DateTime, right? In that case, we can return the MinValue, no? 
```suggestion
            return lastWriteTimeUtc ?? DateTime.MinValue;
```
That should behave as the normal (non-symlink) path.

Reply:
It just compares that previous and current dates are not the same (it doesn't matter in which direction):

![image](https://user-images.githubusercontent.com/6609929/129461377-343765bd-d3e7-42a1-b672-d730e40cf79e.png)

But if you think. returning default or min date would be fine, I'll update the PR. Didn't really dig into that logic, just tried to minimize side effects.

Reply:
Yes please do, MinDate should be best for avoid side effects.

Reply:
Updated the PR